File: src/main/java/org/blacksun/view/MainFrame.java
package org.blacksun.view;

import org.blacksun.network.Network;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;

public class MainFrame extends JFrame {
    public MainFrame(@NotNull Network network) {
        this("NetworkCW", network);
    }

    public MainFrame(String title, @NotNull Network network) {
        super(title);
        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
        Config cfg = Config.getConfig();
        setPreferredSize(new Dimension(cfg.getFrameWidth(), cfg.getFrameHeight()));
        JSplitPane pane = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT);
        NetworkPanel networkPanel = new NetworkPanel(network);
        pane.setLeftComponent(networkPanel);
        JPanel panel = new JPanel();
        panel.add(new ToolbarPanel(networkPanel));
        pane.setRightComponent(panel);
        pane.setOneTouchExpandable(true);
        setContentPane(pane);
        setVisible(true);
        pack();
    }
}

File: src/main/java/org/blacksun/view/MainFrame.kt
package org.blacksun.view

import org.blacksun.network.Network
import org.blacksun.utils.Config

import javax.swing.*
import java.awt.*

class MainFrame(title: String, network: Network) : JFrame(title) {
    constructor(network: Network) : this("NetworkCW", network)

    init {
        defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE
        val cfg = Config.config
        preferredSize = Dimension(cfg.getInt("frameW"),
                cfg.getInt("frameH"))
        val pane = JSplitPane(JSplitPane.HORIZONTAL_SPLIT)
        val networkPanel = NetworkPanel(network)
        pane.leftComponent = networkPanel
        val panel = JPanel()
        panel.add(ToolbarPanel(networkPanel))
        pane.rightComponent = panel
        pane.isOneTouchExpandable = true
        contentPane = pane
        isVisible = true
        pack()
    }
}
